app/scripts/views/choose_what_to_sync.js

```diff
@@ -0,0 +1,116 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'cocktail',
+  'lib/promise',
+  'stache!templates/choose_what_to_sync',
+  'views/form',
+  'views/mixins/back-mixin',
+  'views/mixins/checkbox-mixin',
+  'views/mixins/signup-success-mixin',
+  'underscore'
+],
+function (Cocktail, p, Template, FormView, BackMixin, CheckboxMixin,
+  SignupSuccessMixin, _) {
+  'use strict';
+
+  var View = FormView.extend({
+    template: Template,
+    className: 'choose-what-to-sync',
+
+    initialize: function () {
+      // Account data is passed in from sign up flow.
+      var data = this.ephemeralData();
+      this._account = data && this.user.initAccount(data.account);
+    },
+
+    getAccount: function () {
+      return this._account;
+    },
+
+    beforeRender: function () {
+      // user cannot proceed if they have not initiated a sign up/in.
+      if (! this.getAccount().get('sessionToken')) {
+        this.navigate('signup');
+        return false;
+      }
+    },
+
+    isEngineSupported: function (engineName) {
```

Comment:
Make this a private function and move it below with the rest of the private functions, add a test for it.